includes/express-checkout/class-wc-payments-express-checkout-button-handler.php

```diff
@@ -168,6 +168,18 @@ public function scripts() {
 
 		wp_set_script_translations( 'WCPAY_EXPRESS_CHECKOUT_ECE', 'woocommerce-payments' );
 
+		$wcpay_config = rawurlencode( wp_json_encode( WC_Payments::get_wc_payments_checkout()->get_payment_fields_js_config() ) );
+
+		if ( ! wp_script_is( 'WCPAY_WOOPAY_EXPRESS_BUTTON', 'enqueued' ) ) {
+			wp_add_inline_script(
+				'WCPAY_EXPRESS_CHECKOUT_ECE',
+				"
+				var wcpayConfig = wcpayConfig || JSON.parse( decodeURIComponent( '" . esc_js( $wcpay_config ) . "' ) );
```

Comment:
The `wcpayConfig` object [is needed](https://github.com/Automattic/woocommerce-payments/blob/trunk/client/express-checkout/blocks/index.js#L25) in the `expressCheckoutElementPaymentMethod` component. That's why it worked with WooPay given that the object is [exposed](https://github.com/Automattic/woocommerce-payments/blob/f4804ab3dc2b145c66ed646758fdd9d88075949b/includes/class-wc-payments-woopay-button-handler.php#L178-L184) when WooPay was enabled.

Reply:
We can remove this `if` block since the `wcpayConfig` variable declaration already handles the possibility of it being set, with `var wcpayConfig = wcpayConfig || ....`.

Reply:
@rafaelzaleski the problem is that the variable is printed twice in the frontend.

Reply:
That's a valid point. I also feel uncomfortable having a condition related to WooPay within the ECE code, as they are distinct entities.

However, we could remove the `wcpayConfig` dependency in `expressCheckoutElementPaymentMethod` because the component won’t even be registered if the ECE feature flag is disabled (see [this code](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/blocks/index.js#L157C21-L157C52)).

```
[...]
const expressCheckoutElementPaymentMethod = ( api ) => ( {
[...]
canMakePayment: () => {
		if ( typeof wcpayExpressCheckoutParams === 'undefined' ) {
			return false;
		}

		return true;
	},
[...]
```

If we remove the dependency, setting a minimum height for the container might be enough. What do you think?

Reply:
That's a great idea. I've update the code.

Reply:
@rafaelzaleski Done [a3760ec](https://github.com/Automattic/woocommerce-payments/pull/8905/commits/a3760ecaed634c502df1cf1b268085b6482fd663)